File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans = "" ;
        for(int i = 0 ; i < strs[0].size() ; i++){
            char c = strs[0][i] ; 
            bool flag = 1 ;
            for(int j = 0 ; j < strs.size() ; j++){
                if(strs[j][i] != c)
                    flag = 0;
            }
            if(flag == 1)
                ans += c ;
            else 
                break ; 
        } 
        return ans ;  
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String ans = "" ;
        int flag= 1 ;
        for(int i = 0 ; i < strs[0].length() ; i++){
            ans += strs[0].charAt(i) ;
            for(String str : strs){
                if(!str.startsWith(ans)){
                    flag = -1 ;
                    break ;
                }
            }
            if(flag == -1){
                ans = ans.substring(0,ans.length() - 1) ;
                break;
            }

        }
        return ans ;
    }
}